### ISSUE TITLE
Add documentation needed for a users who are adding new devices to the Cirq

### ISSUE THREAD

**Issue by karlunho** [quantumlib/Cirq#4175]
New hardware vendors need some documentation 

**Describe the solution you'd like**

Content:
- How to do gate composition
- How to add gate sets
-  

**[optional] Describe alternatives/workarounds you've considered**

**[optional] Additional context (e.g. screenshots)**

**What is the urgency from your perspective for this issue? Is it blocking important work?**

P0 - this should land no later than a week
P1 - I need this no later than the next release (end of quarter)
P2 - we should do it in the next couple of quarters 
P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

**Comment by vtomole** [comment]
This is a duplicate of https://github.com/quantumlib/Cirq/issues/3820.
